Store empty string if embed does not have thumbnail_url

cnk opened the pull request:
Fixes issue #6788 

* Do the tests still pass? yes
* Does the code comply with the style guide? yes
* For Python changes: Have you added tests to cover the new/fixed behaviour? Modified one existing test to cover this
* For front-end changes: Did you test on all of Wagtail’s supported browsers? N/A
* For new features: Has the documentation been updated accordingly?  N/A

ababic commented:
@cnk how do you feel about also adding a data migration to turn blank string values back to `None`? Not strictly necessary, but I think it would be have consistent falsey values at the DB level.

cnk commented:
@ababic I like the idea of consistent values but looking at my existing data, it looks like the consistent way to do this would be to keep the not null but default the value to ''. I'll looks some more and amend this PR.

allcaps commented:
If we want consistency for empty values, I agree with @cnk and suggest to migrate to empty string. The oEmbed providers who provide Null values should be cast to empty string on save too.

In this case, from the Django field perspective, there isn't any good reason to store Null.

pySilver commented:
+1 for @allcaps - most of the time nullable char fields brings nothing but the pain.

pySilver commented on wagtail/embeds/embeds.py:
```diff
@@ -42,6 +42,10 @@ def finder(url, max_width=None):
     if 'html' not in embed_dict or not embed_dict['html']:
```
I guess its matter of taste but you can shortcut this (as well as `thumbnail_url` check/conversion) to smth like

```python
embed_dict['html'] = embed_dict.get('html') if isinstance(embed_dict.get('html'), str) else ''
embed_dict['thumbnail_url'] = embed_dict.get('thumbnail_url') if isinstance(embed_dict.get('thumbnail_url'), str) else ''
```

but again, thats just matter of taste. maybe your variant is more readable. 

you can always simply check if value is a string, because I guess we don't care if its _falsey_  (empty string, False, None, 0 etc) but rather if its a string type.

cnk replied:
Sorry I didn't see this before I pushed my (hopefully) last commit on this.

cnk commented:
@allcaps @ababic OK I have added a data migration to convert null thumbnail_urls to '' and updated a couple of tests to demonstrate the behavior when thumbnail_url is missing in the data from the embed provider

ababic commented on wagtail/embeds/migrations/0009_convert_null_to_empty_string.py:
```diff
@@ -0,0 +1,18 @@
+
+    operations = [
+        migrations.RunPython(migrate_forwards, migrations.RunPython.noop),
+    ]
```
Hi @cnk. Thank you very much for this, but I think this migration might be redundant. I could have probably done a better job of outlining that we would only need this if adding `null=True` back to the field... The AlterField operation in the previous migration should have already taken care of converting NULL values to blank strings.

gasman reviewed (approved):
Thanks @cnk - code looks good to me. Have just checked that on both postgresql and sqlite, the existing migration for adding the NOT NULL constraint does indeed change nulls to empty strings as per @ababic's comment, so the new migration isn't necessary - will go ahead and merge without the migration.

gasman commented:
Merged in afc3349c94ea3756f25b6444eea585ebd0c02fdf (master) and d0e6d230b21ce9ca65dd28d93022b01b90cffc6a (stable/2.12.x).
